Lib/fontTools/ttLib/tables/_g_l_y_f.py

```diff
@@ -243,6 +243,154 @@ def __len__(self):
+				# of the default master as the vertical origin
+				assert defaultVerticalOrigin is not None
+				topSideY = defaultVerticalOrigin
+		bottomSideY = topSideY - verticalAdvanceWidth
+		return [
+			(leftSideX, 0),
+			(rightSideX, 0),
+			(0, topSideY),
+			(0, bottomSideY),
+		]
+
+	def getCoordinatesAndControls(self, glyphName, ttFont, defaultVerticalOrigin=None):
+		"""Return glyph coordinates and controls as expected by "gvar" table.
+
+		The coordinates includes four "phantom points" for the glyph metrics,
+		as mandated by the "gvar" spec.
+
+		The glyph controls is a namedtuple with the following attributes:
+			- numberOfContours: -1 for composite glyphs.
+			- endPts: list of indices of end points for each contour in simple
+			glyphs, or component indices in composite glyphs (used for IUP
+			optimization).
+			- flags: array of contour point flags for simple glyphs (None for
+			composite glyphs).
+			- components: list of base glyph names (str) for each component in
+			composite glyphs (None for simple glyphs).
+
+		The "ttFont" and "defaultVerticalOrigin" args are used to compute the
+		"phantom points" (see "getPhantomPoints" method).
+
+		Return None if the requested glyphName is not present.
+		"""
+		if glyphName not in self.glyphs:
+			return None
+		glyph = self[glyphName]
+		if glyph.isComposite():
+			coords = GlyphCoordinates(
+				[(getattr(c, 'x', 0), getattr(c, 'y', 0)) for c in glyph.components]
+			)
+			controls = _GlyphControls(
+				numberOfContours=glyph.numberOfContours,
+				endPts=list(range(len(glyph.components))),
+				flags=None,
+				components=[c.glyphName for c in glyph.components],
+			)
+		else:
+			coords, endPts, flags = glyph.getCoordinates(self)
+			coords = coords.copy()
+			controls = _GlyphControls(
+				numberOfContours=glyph.numberOfContours,
+				endPts=endPts,
+				flags=flags,
+				components=None,
+			)
+		# Add phantom points for (left, right, top, bottom) positions.
+		phantomPoints = self.getPhantomPoints(
+			glyphName, ttFont, defaultVerticalOrigin=defaultVerticalOrigin
+		)
+		coords.extend(phantomPoints)
+		return coords, controls
+
+	def setCoordinates(self, glyphName, coord, ttFont):
+		"""Set coordinates and metrics for the given glyph.
+
+		"coord" is an array of GlyphCoordinates which must include the four
+		"phantom points".
+
+		Both the horizontal/vertical advances and left/top sidebearings in "hmtx"
+		and "vmtx" tables (if any) are updated from four phantom points and
+		the glyph's bounding boxes.
+		"""
+		# TODO: Create new glyph if not already present
+		assert glyphName in self.glyphs
+		glyph = self[glyphName]
+
+		# Handle phantom points for (left, right, top, bottom) positions.
+		assert len(coord) >= 4
+		leftSideX = coord[-4][0]
+		rightSideX = coord[-3][0]
+		topSideY = coord[-2][1]
+		bottomSideY = coord[-1][1]
+
+		coord = coord[:-4]
+
+		if glyph.isComposite():
+			assert len(coord) == len(glyph.components)
+			for p,comp in zip(coord, glyph.components):
```

Comment:
A nit: the rest of file appears to have a space after a comma.  That's missing here and two lines below.

Reply:
it was copy-pasted from elsewhere, but i'll fix that, thanks.